hellogsm-web/src/main/java/team/themoment/hellogsm/web/domain/application/service/impl/DeleteApplicationServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package team.themoment.hellogsm.web.domain.application.service.impl;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import team.themoment.hellogsm.web.domain.application.repository.ApplicationRepository;
+import team.themoment.hellogsm.web.domain.application.service.DeleteApplicationService;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
```

Comment:
```suggestion
@Transactional(rollbackFor = {Exception.class})
```

`@Transactional`의 기본적인 롤백 전략은 unchecked exception라서 checked exception이 발생했을 때, 롤백되지 않을 수 있습니다. 
자바에선 checked exception은 뒤에 throws로 알려주긴 합니다. 
다만 굳이 의도적으로 checked exception을 날려서 관리를 까다롭게 할 계획은 없어서 `rollbackFor` 옵션을 지정해서 모든 예외를 처리하도록 하고 있습니다.

Reply:
https://da-nyee.github.io/posts/spring-transactional-rollbackfor-exceptionclass

관련한 내용을 다루는 글인데, 한번 읽어보시면 좋아요.